File: adventofcode-2024/src/main/java/nl/mvdr/adventofcode/adventofcode2024/day20/Part1.java
package nl.mvdr.adventofcode.adventofcode2024.day20;

import java.util.stream.Stream;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import nl.mvdr.adventofcode.solver.LongSolver;

public class Part1 implements LongSolver {

    private static final Logger LOGGER = LoggerFactory.getLogger(Part1.class);

    private final int minSavings;
    
    Part1(int minSavings) {
        this.minSavings = minSavings;
    }
    
    public Part1() {
        this(100);
    }
    
    @Override
    public long solve(Stream<String> lines) {
        return new RaceConditionSolver(2, minSavings).solve(lines);
    }
    
    public static void main(String[] args) {
        var instance = new Part1();

        var result = instance.solve("input-day20-2024.txt");

        LOGGER.info(result);
    }
}
 

File: adventofcode-2025/src/main/kotlin/nl/mvdr/adventofcode/adventofcode2025/day10/Part1.kt
package nl.mvdr.adventofcode.adventofcode2025.day10

import io.github.oshai.kotlinlogging.KotlinLogging
import nl.mvdr.adventofcode.adventofcode2025.day08.findCircuits
import nl.mvdr.adventofcode.adventofcode2025.day08.findPairs
import nl.mvdr.adventofcode.adventofcode2025.day08.parseJunctionBoxes
import nl.mvdr.adventofcode.solver.FunctionSolver

private val logger = KotlinLogging.logger{}

fun solvePart1(lines: Sequence<String>) = lines.map(::parseMachine).sumOf(Machine::computeFewestButtonPressesToInit)

fun main() {
    val result = FunctionSolver(::solvePart1).solve("input-day10.txt")
    logger.info { result }
}
